addons/sale/tests/test_sale_to_invoice.py

```diff
@@ -100,7 +101,7 @@ def test_downpayment(self):
         self.assertEqual(len(self.sale_order.invoice_ids), 2, 'Invoice should be created for the SO')
 
         invoice = self.sale_order.invoice_ids.sorted()[-1]
-        self.assertEqual(len(invoice.invoice_line_ids), len(self.sale_order.order_line), 'All lines should be invoiced')
+        self.assertEqual(len(invoice.invoice_line_ids.filtered(lambda l: not (l.display_type == 'line_section' and l.name == _('Down Payments')))), len(self.sale_order.order_line), 'All lines should be invoiced')
```

Comment:
Filtering on a translatable field is probably not a good idea, but lang should stay the same in the context of this test. Don't you have other means to find the downpayment line ? If hackish just let as it is.

Reply:
The business need is that when they are downpayment they should be placed in a "Down Payment" section. This unfortunately adds a line in the invoice and the comparison that is made is getting wrong if we do not remove the "Down Payment" section(s). As those are sections like any other I do not see how to distinguish them in another way than using their label... so I leave it that way

For the translation I remove it as advised.